src/app/features/course-info/course-info.component.ts

```diff
@@ -1,20 +1,52 @@
-import { Component, Input } from "@angular/core";
+import { Component, EventEmitter, OnInit } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
 import { ButtonLabels } from "@app/shared/constants/button-labels";
-interface Course {
-  id: string;
-  title: string;
-  description: string;
-  creationDate: string;
-  duration: number;
-  authors: string[];
-}
+import { mockedCoursesList } from "@app/shared/mocks/mocks";
+import { Location } from "@angular/common";
+import { CoursesStoreService } from "@app/services/courses-store.service";
+import { Course } from "@app/shared/models/course.model";
+
 @Component({
   selector: "app-course-info",
   templateUrl: "./course-info.component.html",
   styleUrls: ["./course-info.component.scss"],
 })
-export class CourseInfoComponent {
-  @Input() course!: Course;
+export class CourseInfoComponent implements OnInit {
+  course!: Course | undefined;
   btnTextBack: string = ButtonLabels.back;
-  // Use the names for the input `course`.
+  courseId: string | null = null;
```

Comment:
Keeping track of the id is not necessary because it is property of the course object. Pls refactor to handle id only in block scopes.